learn/work/sparse-recipe/index.qmd

```diff
@@ -0,0 +1,254 @@
+}
+
+flights <- flights |>
+  mutate(across(c(dep_time, sched_arr_time), get_minutes))
+```
+
+We will fit a model using the first month of the year, and then try to assess how well it will generalize over the remaining years. We will also exclude any observations where `arr_delay` is `NA`.
+
+```{r}
+flights <- drop_na(flights, arr_delay)
+flights_train <- filter(flights, month == 1)
+flights_test <- filter(flights, month != 1)
+```
+
+## Creating a recipe
+
+The data is quite simple in terms of types. We have numeric variables and categorical variables. We will do some simple imputation of the numeric variables and create dummy variables on the categorical predictors.
+
+```{r}
+rec_spec <- recipe(arr_delay ~ ., data = flights_train) |>
+  step_impute_mean(all_numeric_predictors()) |>
+  step_zv(all_predictors()) |>
+  step_normalize(all_numeric_predictors()) |>
+  step_novel(all_nominal_predictors()) |>
+  step_unknown(all_nominal_predictors()) |>
+  step_dummy(all_nominal_predictors())
+
+rec_spec
+```
+
+You will notice that we aren't doing anything special here to denote the recipe to produce sparsity. Next will go into some details to explain how you the user should approach recipes when you suspect that sparsity will be produced.
+
+## How is sparsity handled in recipes
+
+There have been made two types of changes to recipes steps with regards to sparsity. First, a number of steps have gained a `sparse` argument. This is used to toggle the creation of sparse vectors. `step_dummy()` that we used in our recipe above is one of such steps.
+
+The second change is that a number of steps are now able to take sparse vectors as input while preserving sparsity. You can see a full [list of these steps](../../../find/sparse/index.qmd) at the link.
+
+Most of the changes with regard to sparsity are done to minimize the changes the user needs to make to their code. This means that in many cases you don't need to change anything, the steps will know when to produce sparse data or not.
+
+When a recipe is used in a workflow and it is being `fit()`, an internal check is being done to figure out whether or not to produce sparse features. This check looks at the sparsity of the data itself, what model is being used, and the recipe. Since [only some models](../../../find/sparse/index.qmd) support sparsity this is the first check. 
+
+A rough estimate of the sparsity of the data that will come out of the recipe is calculated. This is done using the input data set, and the steps present in the recipe. But since this check has to happen before the recipe is prepped, it will be quite simple. What this means in practice is that it is fairly good at estimating the sparsity that is produced by sparsity-generating steps, but it isn't able to detect if those variables are passed to a different step that doesn't preserve it.
+
+The following recipe would be accurately estimated as the dummies produced by `step_dummy()` aren't passed to any other steps.
```

Comment:
I'm not sure what "accurately estimated" was intended to convey. Do you mean we make the right decision on the conversion?

Reply:
covered in https://github.com/tidymodels/tidymodels.org/pull/97/commits/e94bc6c0ce99a9bf2c751d37503cf2431118e15b